cni/pkg/taint/README.md

```diff
@@ -0,0 +1,70 @@
+# istio node readiness gateway 
+This package is used to add readiness taint to prevent race condition during critical daemonset pod installation
+## How it works
+It will load the configmap defined by the user which contains the information of critical labels and their namespaces
+Controller will monitoring on all nodes and pods with critical labels and namespaces defined in the configuration map
+it will taint the node if some of the critical labels are not setup in current node
+and when all critical labels in given node is set up. it will untaint the node to allow non-critical pods to register
+## What it will do and not do?
+1. It is a complementary package to repair controller because repair controller itself cannot prevent daemonset failure and when istio-cni daemonset becomes unready, it is not able to install iptable rules to pods and introduce race condition.
+thus it **must work together with istio-cni-repair controller**
+2. It support much more generalized setting in node readiness checking, thus user can define their own configuration maps for more complicated readiness Check
+## How to use
+### create a configmap to define critical labels and their namespace
+Configmap defines the namespace and label selector for critical pod, and in default it should be located in kube-system namespace with name node.readiness to let controller find them automatically
+An example of configmap
+Layout
+```bash
+./
+configs/
+	Istio-cni.properties
+```
+```bash
+Istio-cni.properties file
+
+name=istio-cni
+selector=app=istio
+namespace=kube-system
+```
+command to create the configmap
+```bash
+
+kubectl create configmap node.readiness -n kube-system --from-file=./configs/
```

Comment:
nit: don't use `.` in the name

Reply:
Also can you just post the configmap output, I think it is preferred to the imperative commands